ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fullsync/FullSyncTargetManager.java

```diff
@@ -105,6 +108,6 @@ private boolean isSyncTargetReached(final EthPeer peer) {
 
   @Override
   public boolean shouldContinueDownloading() {
-    return true;
+    return terminationCondition.shouldContinueDownload();
```

Comment:
merging this pr onto a devnet4 branch, I am still seeing the pipeline run all the way to the head of peers (with no-op imports).  But when I restart the client, it correctly does not try to sync beyond TTD.  It seems like we are not evaluating this during a running pipeline.

Reply:
The pipeline should now be stopping properly when the terminal condition is reached.